test/unit/video.test.js

```diff
@@ -603,3 +606,131 @@ QUnit.test('adds video-js class name with the video-js embed', function(assert)
   assert.ok(player.hasClass('video-js'), 'video-js class was added to the first embed');
   assert.ok(player2.hasClass('video-js'), 'video-js class was preserved to the second embed');
 });
+
+QUnit.module('videojs integration');
+/**
+ * This test is very important for dom-data memory checking
+ * as it runs through a basic player lifecycle for real.
+ */
+QUnit.test('create a real player and play', function(assert) {
+  assert.timeout(30000);
+  const done = assert.async();
+  const fixture = document.getElementById('qunit-fixture');
+  const old = {};
+
+  // TODO: remove this code when we have a videojs debug build
+  // see https://github.com/videojs/video.js/issues/5858
```

Comment:
Technically this doesn't cover all the cases where we will use wrappers for event listeners, but for now it would have caught the bug that we were seeing.